File: 2022/Python/day_1.py
def parse_input():
    splitter = []
    with open("Inputs/day_1_input.txt") as input_file:
        for line in input_file:
            ltext = line.replace("\n", "")
            splitter.append(ltext)
    return splitter


def part_1():
    data = parse_input()
    fatso = 0
    cals = 0
    elf_counter = 1
    elf_tracker = 0

    for i in data:
        if i == "":
            if cals > fatso:
                fatso = cals
                elf_tracker = elf_counter
            elf_counter += 1
            cals = 0
        else:
            cals += int(i)

    print(
        "Part 1: The fattest elf is Elf # "
        + str(elf_tracker)
        + " at "
        + str(fatso)
        + " calories."
    )


def part_2():
    data = parse_input()
    elfs_data = []
    cals = 0
    elf_counter = 1

    for i in data:
        if i == "":
            elfs_data.append([cals])
            elf_counter += 1
            cals = 0
        else:
            cals += int(i)

    elfs_data.sort(reverse=True)
    three_fat = int(elfs_data[0][0]) + int(elfs_data[1][0]) + int(elfs_data[2][0])

    print("Part 2: The 3 fattest elves calories total: " + str(three_fat))


if __name__ == "__main__":
    part_1()
    part_2()

File: 2022/C/Day_1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

void part_1();
void part_2();
void check_arr(int elf, int *arr);

int main() {
  part_1();
  part_2();
}

void part_1() {
  FILE *file = fopen("inputs/day_1_input.txt", "r");
  char line[1000];
  int fattest = 0;
  int cals = 0;

  while (fgets(line, sizeof(line), file) != NULL) {
    int line_cals = atoi(line);

    if (line_cals == 0) {
      if (cals > fattest) {
        fattest = cals;
      }
      cals = 0;
    } else {
      cals += line_cals;
    }
  }
  fclose(file);

  printf("The total cals for the fattest elf is: %d\n", fattest);
}

void part_2() {
  FILE *file = fopen("inputs/day_1_input.txt", "r");
  char line[1000];
  int fattest[3] = {
      1,
      2,
      3,
  };
  int cals = 0;

  while (fgets(line, sizeof(line), file) != NULL) {
    int line_cals = atoi(line);

    if (line_cals == 0) {
      check_arr(cals, fattest);
      cals = 0;
    } else {
      cals += line_cals;
    }
  }
  fclose(file);

  int final_three = fattest[0] + fattest[1] + fattest[2];
  printf("The total cals for the fattest elf is: %d\n", final_three);
}

void check_arr(int elf, int *arr) {
  if (elf > arr[2]) {
    arr[0] = arr[1];
    arr[1] = arr[2];
    arr[2] = elf;
  } else if (elf > arr[1]) {
    arr[0] = arr[1];
    arr[1] = elf;
  } else if (elf > arr[0]) {
    arr[0] = elf;
  }
}
